test/e2e/app-dir/actions/app-action-size-limit-invalid.test.ts

```diff
@@ -117,6 +117,42 @@ createNextDescribe(
             : ''
         }, 'yes')
       })
+
+      it('should respect the size set in serverActions.bodySizeLimit when submitting form', async function () {
+        await next.patchFile(
+          'next.config.js',
+          `
+      module.exports = {
+        experimental: {
+          serverActions: { bodySizeLimit: '2mb' }
+        },
+      }
+      `
+        )
+        await next.stop()
+        await next.build()
+        await next.start()
+
+        const logs: string[] = []
+        next.on('stdout', (log) => {
+          logs.push(log)
+        })
+        next.on('stderr', (log) => {
+          logs.push(log)
+        })
+
+        const browser = await next.browser('/form')
+        await browser.elementByCss('#size-1mb').click()
+
+        await check(() => {
+          return logs.some((log) => log.includes('size = 1048576')) ? 'yes' : ''
+        }, 'yes')
+
+        await browser.elementByCss('#size-2mb').click()
+        await check(() => {
+          return logs.some((log) => log.includes('size = 2097152')) ? 'yes' : ''
+        }, 'yes')
```

Comment:
How about the 3mb one?

Reply:
When clicking the 3mb I think it said `size = 2097152` since it is truncating the payload.. 

I guess it should throw instead, but busboy does not support that.